awslambda-extension-test/src/test/java/org/ballerinax/awslambda/test/utils/SyntaxTreeJSONGenerator.java

```diff
@@ -0,0 +1,203 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinax.awslambda.test.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.ballerina.compiler.internal.parser.BallerinaParser;
+import io.ballerina.compiler.internal.parser.ParserFactory;
+import io.ballerina.compiler.internal.parser.ParserRuleContext;
+import io.ballerina.compiler.internal.parser.tree.STInvalidNodeMinutiae;
+import io.ballerina.compiler.internal.parser.tree.STMinutiae;
+import io.ballerina.compiler.internal.parser.tree.STNode;
+import io.ballerina.compiler.internal.parser.tree.STNodeDiagnostic;
+import io.ballerina.compiler.internal.parser.tree.STNodeList;
+import io.ballerina.compiler.internal.parser.tree.STToken;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.CHILDREN_FIELD;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.DIAGNOSTICS_FIELD;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.HAS_DIAGNOSTICS;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.INVALID_NODE_FIELD;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.IS_MISSING_FIELD;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.KIND_FIELD;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.LEADING_MINUTIAE;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.TRAILING_MINUTIAE;
+import static org.ballerinax.awslambda.test.utils.ParserTestConstants.VALUE_FIELD;
+
+/**
+ * Generates a JSON that represents the structure of the syntax tree. This JSON
+ * can be used to validate the parsed tree during unit-tests.
+ *
+ * @since 1.2.0
+ */
+public class SyntaxTreeJSONGenerator {
+
+    /*
+     * Change the below two constants as required, depending on the type of test.
+     */
+    private static final ParserRuleContext PARSER_CONTEXT = ParserRuleContext.COMP_UNIT;
+
+    private static final PrintStream STANDARD_OUT = System.out;
+    private static final Path RESOURCE_DIRECTORY = Paths.get("src/test/resources/handlers/generated");
+
+    public static void main(String[] args) throws IOException {
+        // Using a string source as input
+        // generateJSON("a 7", ParserRuleContext.EXPRESSION);
+
+        // Using a file source as input
+        String path = "generated.bal";
+        String jsonString = generateJSON(Paths.get(path), PARSER_CONTEXT);
+        STANDARD_OUT.println(jsonString);
```

Comment:
Do we need to print the output?

Reply:
yes. this is for generating the json tree based on a .bal file. we only use this for generating test artifacts.